app/src/main/java/org/mozilla/fenix/session/OnRegisterActLifeCycleCallback.kt

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.session
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import org.mozilla.fenix.components.Performance
+
+/**
+ * This ActivityLifecycleCallbacks implementations tracks if a frame is visible and
+ * allows us to start a queue of tasks that have been delayed until after visual completeness.
+ */
+@SuppressWarnings("EmptyFunctionBlock")
+class OnRegisterActLifeCycleCallback(private val performance: Performance) :
```

Comment:
nit: typically in a dependency injection pattern, we'd pass in the smallest component needed by the class: in this case, the `startupTaskManager` rather than `Performance`.